### ISSUE TITLE
Fastest config for v1.1

### ISSUE THREAD

**Issue by ivysrono** [lightvector/KataGo#29]
Notice: This is fastest only, not the best.
`CPU`: Intel i5-8500
`GPU`: nVidia RTX 2060
`gtp_example.cfg`:
```
numSearchThreads = 36
nnMaxBatchSize = 36
cudaUseFP16 = true
cudaUseNHWC = true
```

about 2500 visits / second

**Comment by lightvector** [comment]
Neat! Keep in mind that the more threads there are, the weaker the search will be *per visit*. So you will get more visits, but those visits will be slightly less effective, because MCTS benefits from knowing the results of the previous visits when choosing how to spend the next ones.

This means that if you're at the point where numSearchThreads is maximizing visits/second - where increasing numSearchThreads further actually starts to decrease visits/second instead of increase it - then you're probably slightly too high, because at the point where the marginal +1 thread no longer helps performance much, it will still be decreasing strength per visit a little.

**Comment by Splee99** [comment]
What is the relationship of numNNServerThreadsPerModel and numSearchThreads to the hardware, such as number of CPU core and number of computing unit in GPU?

**Comment by lightvector** [comment]
@Splee99 - see https://github.com/lightvector/KataGo/issues/28 for a description of the parameters. I'll see about adding this information to a readme or something too on the next release.

**Comment by Splee99** [comment]
OK, thanks. Now this maybe a separate issue. My CPU has two cores. No matter how I set those parameters, the CPU usage is always 50%. It's like the second core is not used. I'm not sure if this is a Windows issue. When running leelazero both cores are used so there is no hardware issue. Maybe you want to take a look in the next release.

**Comment by lightvector** [comment]
That's a strange problem. I'm not sure what I would even begin take a look at. With regards to threading, KataGo is pretty unremarkable, using C++ standard library <thread> and nothing else. Sadly, I also don't have a good way to test on Windows quite yet, (very grateful to alreadydone for making something work on Windows well i advance of official support).

But maybe there are still things to try. Can you give some more info? For example, is it literally always exactly 50% when searching, or does it fluctuate? Is it still true if you increase numSearchThreads to a much larger number, much past the number of cores you have? If you check in Windows, did somehow a CPU affinity get set for KataGo in particular that restricts the cores it can use? https://superuser.com/questions/181577/what-is-windows-priority-and-affinity-and-what-advatanges-does-it-provide

If it's fluctuating around 50% rather than actually pinned at 50% exactly, a boring explanation could be that given how KataGo is implemented on the GPU and the particular ratio of CPU to GPU power and latency for the system you have, maybe the CPU is simply most of the time waiting on the GPU, enough so that the average usage is never more than 50%.

**Comment by Splee99** [comment]
You are right that the CPU usage is indeed light, while two cores are used. I guess that's how Katago is implemented and parameter tuning doesn't help.
![CPU](https://user-images.githubusercontent.com/35020997/60307070-ddf9f200-98f7-11e9-857d-e7da4e27d176.png)
